### PR TITLE
Add more events to Node Deployments

### LINKED ISSUE #3189: More events for Node Deployments
**User Story**
I'd like to see more events for Node Deployments (MachineSets and MachineDeployments). Currently only events for Machines are returned.

**Acceptance criteria**
- [x] Add more events to the `ListNodeDeploymentNodesEvents` from MachineSets and MachineDeployments

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it**:
Adds more events to Node Deployments from `MachineSet` and `MachineDeployment`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3189 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by floreks** [comment]
I'll coordinate merge with frontend.

/hold

File: api/hack/run-controller.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ set -x
 cd $(go env GOPATH)/src/github.com/kubermatic/kubermatic/api
 # Deploy a user-cluster/ipam-controller for which we actuallly
 # have a pushed image
-export KUBERMATICCOMMIT="${KUBERMATICCOMMIT:-$(git rev-parse master)}"
+export KUBERMATICCOMMIT="${KUBERMATICCOMMIT:-$(git rev-parse origin/master)}"
```
Comment (floreks) [inline api/hack/run-controller.sh:11]: @alvaroaleman I have found this to be the issue for me as it requires from user to always keep local master branch updated. We've spent quite some time tracking it down with @p0lyn0mial as it was picking up old version for me. Is it ok to change it to use remote as the base?

File: api/pkg/handler/v1/common/conversion.go [hunk 2]
```diff
@@ -28,7 +28,18 @@ func ConvertInternalSSHKeysToExternal(internalKeys []*kubermaticapiv1.UserSSHKey
 
 // ConvertInternalEventToExternal converts Kubernetes Events to Kubermatic ones (used in the API).
 func ConvertInternalEventToExternal(event corev1.Event) apiv1.Event {
-	result := apiv1.Event{
+	switch event.InvolvedObject.Kind {
+	case "Machine":
+		event.InvolvedObject.Kind = "Node"
+		break
+	case "MachineSet":
+		event.InvolvedObject.Kind = "Node Set"
```
Comment (floreks) [inline api/pkg/handler/v1/common/conversion.go:35]: Should we use `NodeSet` or `Node Set` here?

File: api/pkg/handler/v1/node/node.go [hunk 3]
```diff
@@ -688,6 +741,48 @@ func getMachineEvents(ctx context.Context, client ctrlruntimeclient.Client, mach
 	return kubermaticEvents, nil
 }
 
+func getMachineEvents(ctx context.Context, client ctrlruntimeclient.Client, machines []clusterv1alpha1.Machine) ([]apiv1.Event, error) {
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/node/node.go]: could we replace `getMachineEvents `, `getMachineSetEvents ` and `getMachineDeploymentEvents ` with a function that accepts ` []metav1.Object` ? For example:

```
func getEventsFor(ctx context.Context, client ctrlruntimeclient.Client, res []metav1.Object) ([]apiv1.Event, error) {
...
events, err := getEvents(ctx, client, res.GetUID())
...
```

File: api/pkg/handler/v1/node/node.go [hunk 4]
```diff
@@ -352,6 +352,38 @@ func getMachinesForNodeDeployment(ctx context.Context, clusterProvider provider.
 	return machines, nil
 }
 
+func getMachineSetsForNodeDeployment(ctx context.Context, clusterProvider provider.ClusterProvider, userInfo *provider.UserInfo, cluster *v1.Cluster, nodeDeploymentID string) (*clusterv1alpha1.MachineSetList, error) {
+	client, err := clusterProvider.GetClientForCustomerCluster(userInfo, cluster)
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/node/node.go]: return `err` as we wrap (`KubernetesErrorToHTTPError`) in `ListNodeDeploymentNodes`.

File: api/pkg/handler/v1/node/node.go [hunk 5]
```diff
@@ -352,6 +352,38 @@ func getMachinesForNodeDeployment(ctx context.Context, clusterProvider provider.
...
+	}
+
+	machineDeployment := &clusterv1alpha1.MachineDeployment{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: metav1.NamespaceSystem, Name: nodeDeploymentID}, machineDeployment); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/node/node.go]: as above.

File: api/pkg/handler/v1/node/node.go [hunk 6]
```diff
@@ -352,6 +352,38 @@ func getMachinesForNodeDeployment(ctx context.Context, clusterProvider provider.
...
+	if err := client.List(ctx, &ctrlruntimeclient.ListOptions{Namespace: metav1.NamespaceSystem, LabelSelector: labels.SelectorFromSet(machineDeployment.Spec.Selector.MatchLabels)}, machineSets); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+	return machineSets, nil
+}
+
+func getMachineDeploymentForNodeDeployment(ctx context.Context, clusterProvider provider.ClusterProvider, userInfo *provider.UserInfo, cluster *v1.Cluster, nodeDeploymentID string) (*clusterv1alpha1.MachineDeployment, error) {
+	client, err := clusterProvider.GetClientForCustomerCluster(userInfo, cluster)
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/node/node.go]: and here, please check other places too.

File: api/pkg/handler/v1/common/conversion.go (hunk 2 above)
Comment (p0lyn0mial) [reply api/pkg/handler/v1/common/conversion.go:35]: `Node Set` is fine for me.

File: api/pkg/handler/v1/node/node.go (hunk 3 above)
Comment (floreks) [reply api/pkg/handler/v1/node/node.go]: Great! I was looking for a way to use some common interface.

File: api/pkg/handler/v1/node/node.go (hunk 4 above)
Comment (floreks) [reply api/pkg/handler/v1/node/node.go]: Ye, I will update the whole file.

File: api/hack/run-controller.sh (hunk 1 above)
Comment (alvaroaleman) [reply api/hack/run-controller.sh:11]: Yup, very good idea

File: api/pkg/handler/v1/common/conversion.go (hunk 2 above)
Comment (alvaroaleman) [reply api/pkg/handler/v1/common/conversion.go:35]: I believe we call it "NodeDeployment" without a space in the Frontend, so probably keep it consistent?

Comment (p0lyn0mial) [reply api/pkg/handler/v1/common/conversion.go:35]: Screenshot 2019-04-12 at 13 04 06

Comment (p0lyn0mial) [reply api/pkg/handler/v1/common/conversion.go:35]: we call it `Node Deployment`, with the space in the UI.

**Pull request merged**